flang/test/Lower/zero-size.f90

```diff
@@ -1,20 +1,46 @@
 ! RUN: bbc -o - %s | FileCheck %s
 
-! CHECK-LABEL: _QPzero
-subroutine zero(aa)
-  real, dimension(:) :: aa
-  print*, size(aa)
+! CHECK-LABEL: _QPzero1
+subroutine zero1(z)
+  real, dimension(:) :: z
+  print*, size(z), z, ':'
+end
+
+! CHECK-LABEL: _QPzero2
+subroutine zero2
+  type dt
+    integer :: j = 17
+  end type
+  ! CHECK: %[[z:[0-9]*]] = fir.alloca !fir.array<0x!fir.type<_QFzero2Tdt{j:i32}>> {bindc_name = "z", uniq_name = "_QFzero2Ez"}
+  ! CHECK: %[[shape:[0-9]*]] = fir.shape %c0 : (index) -> !fir.shape<1>
+  ! CHECK: fir.embox %[[z]](%[[shape]]) : (!fir.ref<!fir.array<0x!fir.type<_QFzero2Tdt{j:i32}>>>, !fir.shape<1>) -> !fir.box<!fir.array<0x!fir.type<_QFzero2Tdt{j:i32}>>>
```

Comment:
Again, we should probably start making our tests more restrictive to prepare for upstreaming.

Reply:
These new tests are intended to verify zero size array values.  These checks exhaustively cover all of the components used to build those values; the relevant parts of these tests are very small.  (Well, zero size arrays really are nothing much .. :) )